src/main/java/dwbh/api/repositories/internal/JooqVotingRepository.java

```diff
@@ -162,14 +164,57 @@ public boolean hasExpired(UUID votingId) {
     return record.map(Record1::component1).map(JooqVotingRepository::hasExpired).orElse(true);
   }
 
+  @Override
+  public List<Voting> listVotingsGroup(
+      UUID groupId, OffsetDateTime startDate, OffsetDateTime endDate) {
+    return context
+        .select(
+            VotingTableHelper.VOTING_ID, VotingTableHelper.CREATED_AT, VotingTableHelper.AVERAGE)
+        .from(VOTING_TABLE)
+        .where(VotingTableHelper.GROUP_ID.eq(groupId))
+        .and(VotingTableHelper.CREATED_AT.between(startDate, endDate))
+        .fetch(JooqVotingRepository::toVoting);
+  }
+
   private static boolean hasExpired(OffsetDateTime createdAt) {
     return createdAt.plusDays(1).isBefore(OffsetDateTime.now());
   }
 
+  @Override
+  public Voting updateVotingAverage(UUID votingId, Integer average) {
+    Record record =
+        context
+            .update(VOTING_TABLE)
+            .set(VotingTableHelper.AVERAGE, average)
+            .where(VotingTableHelper.VOTING_ID.eq(votingId))
+            .returning(
+                VotingTableHelper.VOTING_ID,
+                VotingTableHelper.CREATED_AT,
+                VotingTableHelper.AVERAGE)
+            .fetchOne();
+
+    return toVoting(record);
+  }
+
+  @Override
+  public Integer calculateVoteAverage(UUID votingId) {
+    Record record =
+        context
+            .select(DSL.avg(VoteTableHelper.SCORE))
+            .from(VOTE_TABLE)
+            .where(VoteTableHelper.VOTING_ID.eq(votingId))
+            .fetchOne();
+
+    return record.get(0) == null
```

Comment:
How about this ?

```java
Optional<Record1<BigDecimal>> record =
        context
            .select(DSL.avg(VoteTableHelper.SCORE))
            .from(VOTE_TABLE)
            .where(VoteTableHelper.VOTING_ID.eq(votingId))
            .fetchOptional();

    return record
        .map(Record1::component1)
        .map(decimal -> decimal.setScale(0, RoundingMode.HALF_UP).intValue())
        .orElse(null);
```

Reply:
Ok. I find hard to think functional all the time :(

Changed